core/domain/question_services.py

```diff
@@ -0,0 +1,205 @@
+        question_data=question.question_data,
+        question_data_schema_version=question.question_data_schema_version,
+        collection_id=question.collection_id,
+        language_code=question.language_code,
+    )
+
+    model.commit(committer_id, commit_message, [{
+        'cmd': CMD_CREATE_NEW,
+        'title': question.title
+        }])
+    return model
+
+
+def add_question(committer_id, question):
+    """Saves a new question.
+    Args:
+        committer_id: str. ID of the committer.
+        question: Question. Question to be saved.
+    """
+    commit_message = (
+        'New question created with title \'%s\'.' % question.title)
+    question_model = _create_question(committer_id, question, commit_message)
+
+    return question_model
+
+
+def delete_question(committer_id, question_id, force_deletion=False):
+    """Deletes the question with the given question_id.
+    Args:
+        committer_id: str. ID of the committer.
+        question_id: str. ID of the question.
+        force_deletion: bool. If true, the question and its history are fully
+            deleted and are unrecoverable. Otherwise, the question and all
+            its history are marked as deleted, but the corresponding models are
+            still retained in the datastore. This last option is the preferred
+            one.
+    """
+    question_model = question_models.QuestionModel.get(question_id)
+    question_model.delete(
+        committer_id, feconf.COMMIT_MESSAGE_QUESTION_DELETED,
+        force_deletion=force_deletion)
+
+
+def get_question_from_model(question_model):
+    """Returns domain object repersenting the given question model.
+
+    Args:
+        question_model: QuestionModel. The question model loaded from the
+            datastore.
+
+    Returns:
+        Question. The domain object representing the question model.
+    """
+    return question_domain.Question(
+        question_model.id, question_model.title, question_model.question_data,
+        question_model.question_data_schema_version,
+        question_model.collection_id, question_model.language_code)
+
+
+def get_question_by_id(question_id):
+    """Returns a domain object representing a question.
+
+    Args:
+        question_id: str. ID of the question.
+
+    Returns:
+        Question or None. The domain object representing a question with the
+        given id, or None if it does not exist.
+    """
+    question_model = question_models.QuestionModel.get(question_id)
+    if question_model:
+        question = get_question_from_model(question_model)
+        return question
+    else:
+        return None
+
+
+def apply_change_list(question_id, change_list):
+    """Applies a changelist to a pristine question and returns the result.
+
+    Args:
+        question_id: str. ID of the given question.
+        change_list: list(dict). A change list to be applied to the given
+            question. Each entry in change_list is a dict that represents an
```

Comment:
No, each entry should be a QuestionChange object.

Reply:
see comment below.

Reply:
Replying below.